website/blog/2022-03-13-2.6.0.md

```diff
@@ -0,0 +1,706 @@
+(
+  @f()
+  class {}
+);
+```
+
+#### Improve printing of comments in type aliases in Flow & TS ([#12268](https://github.com/prettier/prettier/pull/12268) by [@duailibe](https://github.com/duailibe))
+
+For Flow, the comments will now be more aligned to how we print comments in assignments where the right-hand side is an object expression:
+
+<!-- prettier-ignore -->
+```js
+// Input
+type Props = // comment explaining the props
+  {
+    isPlaying: boolean
+  };
+
+// Prettier 2.5
+// comment explaining the props
+type Props = {
+  isPlaying: boolean,
+};
+
+// Prettier 2.6
+type Props =
+  // comment explaining the props
+  {
+    isPlaying: boolean,
+  };
+```
+
+And for TS that comment would not be stable with a second format:
+
+<!-- prettier-ignore -->
+```ts
+// Input
+type Props = // comment explaining the props
+  {
+    isPlaying: boolean
+  };
+
+// Prettier 2.5
+type Props = { // comment explaining the props
+  isPlaying: boolean;
+};
+
+// Prettier 2.5 (2nd format)
+type Props = {
+  // comment explaining the props
+  isPlaying: boolean;
+};
+
+// Prettier 2.6
+type Props =
+  // comment explaining the props
+  {
+    isPlaying: boolean,
+  };
+```
+
+#### Support destructuring private proposal ([#12276](https://github.com/prettier/prettier/pull/12276) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+Support [Stage 2 TC39 proposal destructuring private fields](https://github.com/tc39/proposal-destructuring-private) via [Babel 7.17](https://babeljs.io/blog/2022/02/02/7.17.0).
+
+<!-- prettier-ignore -->
+```jsx
+// Example
+class Foo {
+  constructor() {
+    console.log(this.#x); // => 1
+    const { #x: x } = this;
+    console.log(x); // => 1
+  }
+}
+```
+
+#### Support decorator auto accessors syntax ([#12299](https://github.com/prettier/prettier/pull/12299) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+Support auto accessors syntax that is introduced in [new decorators proposal](https://github.com/tc39/proposal-decorators).
+
+<!-- prettier-ignore -->
+```jsx
+// Example
+@defineElement("my-class")
+class C extends HTMLElement {
+  @reactive accessor clicked = false;
+}
+```
+
+### TypeScript
+
+#### Refactor printing definite assignment assertions ([#12351](https://github.com/prettier/prettier/pull/12351) by [@thorn0](https://github.com/thorn0))
+
+Also now definite assignment assertions are printed even when they aren't followed by type annotations, which is an error in TS but not a parse error.
+
+<!-- prettier-ignore -->
+```ts
+// Input
+let a!;
+
+// Prettier 2.5
+let a;
+
+// Prettier 2.6
+let a!;
+```
+
+#### Print trailing comma for rest elements in tuple types ([#12390](https://github.com/prettier/prettier/pull/12390) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+TypeScript allows to put rest elements in places other than the end of tuple types since [4.2](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-2.html#leadingmiddle-rest-elements-in-tuple-types).
```

Comment:
```suggestion
TypeScript has allows rest elements in tuple types (`...T`) to have normal elements come after them since [TypeScript 4.2](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-2.html#leadingmiddle-rest-elements-in-tuple-types).
```